lens-cube/src/main/java/org/apache/lens/cube/parse/StorageTableResolver.java

```diff
@@ -145,14 +144,17 @@ public void rewriteContext(CubeQueryContext cubeql) throws LensException {
    * @param cubeql
    */
   private void resolveStoragePartitions(CubeQueryContext cubeql) throws LensException {
-    Set<Candidate> candidateList = cubeql.getCandidates();
-    for (Candidate candidate : candidateList) {
+    Iterator<Candidate> candidateIterator = cubeql.getCandidates().iterator();
+    while (candidateIterator.hasNext()) {
+      Candidate candidate = candidateIterator.next();
       boolean isComplete = true;
       for (TimeRange range : cubeql.getTimeRanges()) {
-        isComplete &= candidate.evaluateCompleteness(range, failOnPartialData);
+        isComplete &= candidate.evaluateCompleteness(range, range, failOnPartialData);
       }
       if (!isComplete) {
+        candidateIterator.remove();
```

Comment:
@lavkesh Can we directly remove this candidate ? should we not check if partial data is allowed